common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -149,25 +148,31 @@ public TransportClient createClient(String remoteHost, int remotePort) throws IO
       }
 
       if (cachedClient.isActive()) {
-        logger.trace("Returning cached connection to {}: {}", address, cachedClient);
+        logger.trace("Returning cached connection to {}: {}",
+          cachedClient.getSocketAddress(), cachedClient);
         return cachedClient;
       }
     }
 
     // If we reach here, we don't have an existing connection open. Let's create a new one.
     // Multiple threads might race here to create new connections. Keep only one of them active.
+    final long preResolveHost = System.nanoTime();
+    final InetSocketAddress resolvedAddress = new InetSocketAddress(remoteHost, remotePort);
+    final long hostResolveTimeMs = (System.nanoTime() - preResolveHost) / 1000000;
+    logger.trace("Spent {} ms to resolve {}", hostResolveTimeMs, resolvedAddress);
```

Comment:
I think it's more useful to log a warning if this time is greater than say 2 seconds.